slacker/slack_api.py

```diff
@@ -23,9 +23,19 @@ def __init__(self, token = None):
     # Methods that doesn't require the token to be sent.
     self.__token_unneeded = ['api.test']
 
+    # Method that are destructive, modifies state, or sends messages will not be run in read-only
+    # mode.
+    self.__modifying = ['files.delete', 'chat.memessage', 'chat.postmessage', 'chat.postephemeral']
```

Comment:
It would be good if we could move this to the command itself, maybe we add `is_destructive` method which returns `True` or `False` depending if the command is destructive. Then in the Slack API when calling post we can pass that command instance and call `is_destructive`.

I normally try to avoid keeping a list of something that I know I could potentially forget to update. What do you think?

Reply:
Yeah, I was thinking about that, too. In that case, then the "token unneeded" should also be put on the command base class. Good call.

Reply:
On the other hand, commands can use several API methods so not all they do has to be destructive even if most of it is. Like `files.delete` uses that API method but also `files.list` and it has a `--dry-run` to not do anything. The `SlackAPI` destructive restriction only comes into play when not in dry-run and after the user confirms.

But I guess we can just decide on the command level whether or not a command is destructive, because in read-only mode you don't want to run part of the `files.delete` functionality anyway.